adds net-restrict to cli options (#1646)
hyperledger/besu-docs: docs/public-networks/reference/cli/options.md

:::

### `net-restrict`

<Tabs>

<TabItem value="Syntax" label="Syntax" default>

```bash

> --net-restrict=[<String>,..]

Reviewer: ```suggestion
--net-restrict=<subnet>[,<subnet>,...]
```

```

</TabItem>

<TabItem value="Example" label="Example">

```bash
--net-restrict=192.168.1.0/24,10.0.0.0/8
```

</TabItem>

<TabItem value="Environment variable" label="Environment variable">

```bash

> BESU_NET-RESTRICT=192.168.1.0/24,10.0.0.0/8

Reviewer 2: ```suggestion
BESU_NET_RESTRICT=192.168.1.0/24,10.0.0.0/8
```

```

</TabItem>

<TabItem value="Example configuration file" label="Example configuration file"> 

```bash
net-restrict=["192.168.1.0/24","10.0.0.0/8"]
```

</TabItem>

</Tabs>

> Comma-separated array of allowed IP subnets. Enables node operators to nominate peers by their IP subnets. Default is null: no subnet-based peer permission restrictions will be applied.

Reviewer: ```suggestion
A comma-separated list of allowed IP subnets.
Peers whose IP addresses fall within the specific subnets are granted permission to interact with the node. If not specified, no subnet-based peer permissions are applied.
```

Reword suggestion for consistency with other options and a bit more clarity.
